### ISSUE TITLE
CA1822 should not report a violation when a compilation symbol is not defined

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn-analyzers#5325]
### Analyzer

**Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static`

### Analyzer source

**SDK**: 6.0.100-preview.6.21355.2

### Describe the bug

CA1822 reports a violation when a `ConditionalAttribute` is present and the compilation symbol is not defined.

### Steps To Reproduce

```csharp
        [Conditional("DEBUG")]
        public void SetDebugCookie(object cookie) // CA1822
        {
#if DEBUG
            if (_debugCookies == null)
            {
                _debugCookies = new List<KeyValuePair<int, object>>();
            }
#endif
}
```

### Expected behavior

No violation when a `ConditionalAttribute` is present and the symbol is not defined.

### Actual behavior

CA1822 rule violation.

**Comment by xtqqczze** [comment]
This issue can be worked around by removing the redundant `#if DEBUG` since the `ConditionalAttribute` ensures the method is not compiled.

**Comment by mavasani** [comment]
This is by design - the analyzer only sees the final source in the compilation after all preprocessor symbols have been resolved.
